LeadKit/LeadKit/Extensions/UIImage/UIImage+Alpha.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  UIImage+Alpha.swift
+//  LeadKit
+//
+//  Created by Николай Ашанин on 28.09.16.
+//  Copyright © 2016 Touch Instinct. All rights reserved.
+//
+
+import UIKit
+
+public extension UIImage {
+
+    /**
+     - returns: true if the image has an alpha layer.
+     */
+    public func hasAlpha() -> Bool {
+        let alpha = CGImageGetAlphaInfo(CGImage)
+        switch alpha {
+        case .First, .Last, .PremultipliedFirst, .PremultipliedLast:
+            return true
+        default:
+            return false
+        }
+    }
+
+    /**
+     - returns: a copy of the given image, adding an alpha channel if it doesn't already have one.
+     */
+    public func applyAlpha() -> UIImage? {
+        guard !hasAlpha() else {
+            return self
+        }
+
+        let imageRef = CGImage
+        let width = CGImageGetWidth(imageRef)
+        let height = CGImageGetHeight(imageRef)
+        let colorSpace = CGImageGetColorSpace(imageRef)
+
+        // The bitsPerComponent and bitmapInfo values are hard-coded to prevent an "unsupported parameter combination" error
+        let bitmapInfo = CGBitmapInfo(rawValue: CGBitmapInfo.ByteOrderDefault.rawValue |
+            CGImageAlphaInfo.PremultipliedFirst.rawValue)
+        let offscreenContext = CGBitmapContextCreate(nil, width, height, 8, 0, colorSpace, bitmapInfo.rawValue)
```

Comment:
why not
```swift
init?(data: UnsafeMutableRawPointer?, 
width: Int, 
height: Int, 
bitsPerComponent: Int, 
bytesPerRow: Int, 
space: CGColorSpace, 
bitmapInfo: UInt32)
```
?
https://developer.apple.com/reference/coregraphics/cgcontext/1455939-init